### PR TITLE
Install the calico_rkt plugin with calicoctl node --rkt

### PR DESCRIPTION
**Pull request by alexhersh**
[Rkt Plugin](https://github.com/projectcalico/calico-rkt)

TODO: 
- [x] Fix Node-Start UTs .
- [x] Fix STs .
- [x] Release calico_rkt (necessary for install)
- [x] Edit calico_rkt docs to ref new command.

### CODE REVIEW

File: calico_containers/calico_ctl/node.py [hunk 1]
```diff
@@ -417,11 +435,11 @@ def warn_if_hostname_conflict(ip):
...
-                    "with IP address %s. Calico requires each compute host to "
-                    "have a unique hostname. If this is your first time "
-                    "running 'calicoctl node' on this host, ensure that " \
-                    "another host is not already using the " \
-                    "same hostname."  % (hostname, ip))
+                                "with IP address %s. Calico requires each compute host to "
+                                "have a unique hostname. If this is your first time "
+                                "running 'calicoctl node' on this host, ensure that " \
+                                "another host is not already using the " \
+                                "same hostname."  % (hostname, ip))
```
Comment (TrimBiggs) [inline calico_containers/calico_ctl/node.py]: Nice catch here.  Can you also get rid of the unnecessary `\`'s and fit to 80 chars?

File: calico_containers/calico_ctl/node.py [hunk 2]
```diff
@@ -510,3 +528,28 @@ def install_kubernetes(kubernetes_plugin_dir):
         except OSError:
             print "ERROR: Unable to set permissions on Kubernetes plugin"
             sys.exit(1)
+
+
+def install_rkt(rkt_plugin_dir):
```
Comment (TrimBiggs) [inline calico_containers/calico_ctl/node.py]: This is the exact same function as `install_kubernetes`, think you can commonize it in some way?

File: calico_containers/tests/unit/node_test.py [hunk 3]
```diff
@@ -277,8 +279,9 @@ def test_node_start(self, m_attach_and_stream,
     @patch('calico_ctl.node.warn_if_unknown_ip', autospec=True)
     @patch('calico_ctl.node.warn_if_hostname_conflict', autospec=True)
     @patch('calico_ctl.node.install_kubernetes', autospec=True)
+    @patch('calico_ctl.node.install_rkt', autospec=True)
     def test_node_start_call_backup_kube_directory(
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/node_test.py]: Would be nice to have a test like this for rkt too.

File: calico_containers/calico_ctl/node.py [hunk 4]
```diff
@@ -75,6 +76,11 @@
 KUBERNETES_PLUGIN_DIR = '/usr/libexec/kubernetes/kubelet-plugins/net/exec/calico/'
 KUBERNETES_PLUGIN_DIR_BACKUP = '/etc/kubelet-plugins/calico/'
 
+RKT_PLUGIN_VERSION = 'v0.1.0'
+RKT_BINARY_URL = 'https://github.com/projectcalico/calico-rkt/releases/download/%s/calico_rkt' % RKT_PLUGIN_VERSION
+RKT_PLUGIN_DIR = '/usr/lib/rkt/plugins/net/'
+RKT_CONF_DIR = '/etc/rkt/net.d/'
```
Comment (TrimBiggs) [inline calico_containers/calico_ctl/node.py]: Is this used??

**Comment by TrimBiggs** [comment]
A few comments to check out and those other couple of items on your TODO list, but looking pretty good!

File: calico_containers/calico_ctl/node.py (hunk 4 above)
Comment (alexhersh) [inline calico_containers/calico_ctl/node.py]: I left this in for the possibility of making a default/template conf file as we discussed yesterday

Comment (TrimBiggs) [inline calico_containers/calico_ctl/node.py]: Yes, it would be useful for that, but that is not in this PR (unless you're planning to add it later).

**Comment by alexhersh** [comment]
[calico_rkt reference](https://github.com/projectcalico/calico-rkt/pull/3)

**Comment by TrimBiggs** [comment]
LGTM, but need release of calico-rkt before merging.

**Pull request merged**